### PR TITLE
Testing : Fix vfs pointer leak and use automatic mocking

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

**Comment by loosebazooka** [comment]
@elharo @patflynn 

One of these mocks wasn't working with idea14, so updated it in a way we could have common code in both branches.

Also fixed that pesky vfs pointer not disposed thing.

File: core-plugin/testSrc/com/google/gct/idea/appengine/validation/EndpointTestBase.java [hunk 1]
```diff
@@ -75,4 +86,24 @@ public void run() {
       }
     }
   }
+
+  private void removeEndpointsSdkFromProject() {
+    ApplicationManager.getApplication().runWriteAction(new Runnable() {
+      @Override
+      public void run() {
+        final LibraryTable table =
```
Comment (elharo) [inline core-plugin/testSrc/com/google/gct/idea/appengine/validation/EndpointTestBase.java]: why final? Google style is not to make local varibles final unless required. Is IntelliJ style different?

File: core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java [hunk 2]
```diff
@@ -61,17 +28,10 @@
...
-import java.util.Comparator;
 import java.util.List;
-import java.util.Set;
 
-import javax.swing.Icon;
 import javax.swing.JComponent;
-import javax.swing.event.HyperlinkListener;
 
 
 // TODO: This test hangs when run as part of the full suite.
```
Comment (elharo) [inline core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java]: Is this still true?

**Comment by elharo** [comment]
lgtm

File: core-plugin/testSrc/com/google/gct/idea/appengine/validation/EndpointTestBase.java (hunk 1 above)
Comment (loosebazooka) [inline core-plugin/testSrc/com/google/gct/idea/appengine/validation/EndpointTestBase.java]: There's no reason, I must have just copied it from the "addEndpointsSdk..." code above, will change it back to be consistent.

File: core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java (hunk 2 above)
Comment (loosebazooka) [inline core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java]: I don't think so.

Comment (loosebazooka) [inline core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java]: oh I missed the @ignore here, let me see what's going on.

Comment (loosebazooka) [inline core-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudDebugHistoricalSnapshotsTest.java]: Still broken, so leaving the ignore in. I think we need to use IdeaTestCase or one of the other built ins so they all the state is managed correctly.

**Pull request merged**